src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,48 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 4;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private StorageDao storageDao;
 
     @Override
     public User register(User user) {
-        return null;
+        storageDao = new StorageDaoImpl();
+
+        if (user == null) {
+            throw new NullPointerException("User is null");
+        }
+
+        if (user.getLogin() == null) {
+            throw new NullPointerException("User's login is null");
+        }
+
+        if (user.getPassword() == null) {
+            throw new NullPointerException("User's password is null");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("User with login " + user.getLogin() + " already exists");
+        }
+
```

Comment:
Why not use if-else statements?

Reply:
There is no point in this, RuntimeException interrupts program

Reply:
Yep, but it'd look better, imo. However, it's a matter of taste